azure-common/azure/profiles/__init__.py

```diff
@@ -0,0 +1,127 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#--------------------------------------------------------------------------
+from enum import Enum
+
+class ProfileDefinition(object):
+    """Allow to define a custom Profile definition.
+
+    Note::
+
+    The dict format taken as input is yet to be confirmed and should
+    *not* be considered as stable in the current implementation.
+
+    :param dict profile_dict: A profile dictionnary
+    :param str label: A label for pretty printing
+    """
+    def __init__(self, profile_dict, label=None):
+        self._profile_dict = profile_dict
+        self._label = label
+
+    @property
+    def label(self):
+        """The label associated to this profile definition.
+        """
+        return self._label
+
+    def __repr__(self):
+        return self._label if self._label else self._profile_dict.__repr__()
+
+    def get_profile_dict(self):
+        """Return the current profile dict.
+
+        This is internal information, and content should not be considered stable.
+        """
+        return self._profile_dict
+
+
+class DefaultProfile(object):
+    """Store a default profile.
+
+    :var ProfileDefinition profile: The default profile as class attribute
+    """
+    profile = None
+
+    def use(self, profile):
+        """Define a new default profile."""
+        if not isinstance(profile, KnownProfiles) and not isinstance(profile, ProfileDefinition):
```

Comment:
`isinstance` can take a tuple of classes.
May make it simpler?

Reply:
Yes indeed